xml/schema/CRM/Funding/FundingCasePermissionsCache.xml

```diff
@@ -0,0 +1,70 @@
+<?xml version="1.0" encoding="iso-8859-1" ?>
+
+<table>
+  <base>CRM/Funding</base>
+  <class>FundingCasePermissionsCache</class>
+  <name>civicrm_funding_case_permissions_cache</name>
+  <comment>Cache for FundingCase permissions</comment>
+  <log>true</log>
+
+  <field>
+    <name>id</name>
+    <type>int unsigned</type>
+    <required>true</required>
+    <comment>Unique FundingCasePermissionsCache ID</comment>
+    <html>
+      <type>Number</type>
+    </html>
+  </field>
+  <primaryKey>
+    <name>id</name>
+    <autoincrement>true</autoincrement>
+  </primaryKey>
+
+  <field>
+    <name>funding_case_id</name>
+    <type>int unsigned</type>
+    <required>true</required>
+    <comment>FK to FundingCase</comment>
+  </field>
+  <foreignKey>
+    <name>funding_case_id</name>
+    <table>civicrm_funding_case</table>
+    <key>id</key>
+    <onDelete>CASCADE</onDelete>
+  </foreignKey>
+
+  <field>
+    <name>contact_id</name>
+    <type>int unsigned</type>
+    <required>true</required>
+    <comment>No FK to contact to work with 0 (contact ID on CLI), too</comment>
+  </field>
+
+  <field>
+    <name>is_remote</name>
+    <type>boolean</type>
+    <required>true</required>
+    <comment>Indicates whether the permissions are for internal or remote requests</comment>
+  </field>
+
+  <field>
+    <name>permissions</name>
+    <type>varchar</type>
+    <length>512</length>
```

Comment:
It contains just an array of strings. If each permission has an average length of 20 this would still allow more than 20 permissions per user. I think this should be enough. (It's the same length as in the relation table.)

Reply:
Sounds reasonable for now, but hypothetically infinite-length data blobs should really be `TEXT` instead of `VARCHAR`, don't you think? This will hit us eventually, so we should at least make sure there's validation on all UIs this affects - which seems … out of scale for something like a cache. Are we afraid of too-large tables with a `TEXT` column instead?

Reply:
The question is why a record with contact ID `0` (which doesn't exist) should be persisted by the permissions cache in the first place. Is there a chance of validating so that there's no attempt to write a broken FK to the DB?